qiita_db/sql_connection.py

```diff
@@ -150,16 +150,32 @@ def _open_connection(self):
 
         try:
             self._connection = connect(**args)
-        except Exception:
-            # catch any exception and raise as runtime error
-            # attempt to point the user to their config file
-            error_text = ("Can't connect to database; user `%s` is not known. "
-                          "Your QIITA configuration file, located at `%s` must"
-                          " specify the correct user, via the `USER` parameter"
-                          " under the `[postgres]` header. Please refer to "
-                          "`INSTALL.md` in the QIITA base directory.")
-            error_text = error_text % (qiita_config.user, qiita_config.conf_fp)
-            raise RuntimeError(error_text)
+        except Exception as e:
+            # catch threee known common exceptions and raise runtime errors
+            ebase = ('An error with the following text occurred\n\n\t%s\n%s '
+                     'Please review `INSTALL.md`.')
+            # parse string of expected errors - if too weird, we can't suggest
+            # a solution
+            try:
+                etype = str(e).split(':')[1].split()[0]
```

Comment:
If the raised error didn't include a `:` this would except, so I don't think doing this is a good idea.